ocaml/xapi/xapi_clustering.ml

```diff
@@ -61,17 +61,17 @@ let ip_of_pif (ref,record) =
     }*)
 let assert_pif_prerequisites pif =
   let assert_pif_permaplugged (ref,record) =
-    if not record.API.pIF_disallow_unplug then failwith (Printf.sprintf "PIF %s allows unplug" record.API.pIF_uuid);
-    if not record.pIF_currently_attached then failwith (Printf.sprintf "PIF %s not plugged" record.API.pIF_uuid)
+    if not record.API.pIF_disallow_unplug then
+      raise Api_errors.(Server_error (internal_error, [ "PIF allows unplug"; record.API.pIF_uuid ] ));
```

Comment:
I think we don't need to change this, as exceptions are automatically converted to INTERNAL_ERROR, but it's best to double-check

Reply:
@edwintorok Thoughts?

Reply:
There is already an api error saying that pif does not allow unplug, for symmetry we could introduce an error saying that clustering requires the pif to disallow unplug. pif_allows_unplug.